nexus/db-queries/src/db/datastore/vmm.rs

```diff
@@ -307,6 +305,40 @@ impl DataStore {
             })
     }
 
+    /// Transitions a VMM to the `SagaUnwound` state.
+    ///
+    /// This may *only* be called by the saga that created a VMM record.
```

Comment:
Not a huge thing, but it might be good to mention explicitly that this clobbers the generation number and isn't concurrency-controlled (which is safe because the "only called by an unwinding saga" precondition ensures that no one else, particularly not a sled agent, should be accessing the record concurrently anyway).

Just to check my understanding: I assume doing it this way is the way we're getting around the fact that the unwind steps in the sagas can't check to see if a descendant node was run?

Reply:
> Just to check my understanding: I assume doing it this way is the way we're getting around the fact that the unwind steps in the sagas can't check to see if a descendant node was run?

Yeah, that's correct. I was really hoping there would be a  nicer solution to this, but I couldn't come up with one...

Reply:
I guess an alternative approach could be to have the reverse actions for the
nodes that increment the generation number also attempt to move the  VMM to
SagaUnwound, and then allow the prior nodes attempt to do it to fail because
they were run with a lower generation? I dunno which is less sketchy OTTOMH...

Reply:
The way you have it seems simpler to me.